modules/javafx.base/src/main/java/javafx/beans/value/ObservableValue.java

```diff
@@ -137,4 +139,55 @@
      * @return The current value
      */
     T getValue();
+
+    /**
+     * Returns an {@link ObservableValue} which provides a mapping of the value
+     * held by this {@code ObservableValue}, and provides {@code null} when this
+     * {@code ObservableValue} holds {@code null}.
+     *
+     * @param <U> the type of values held by the resulting {@code ObservableValue}
+     * @param mapper a {@link Function} which converts a given value to a new value, cannot be null
+     * @return an {@link ObservableValue} which provides a mapping of the value
+     *     held by this {@code ObservableValue}, and provides {@code null} when
+     *     this {@code ObservableValue} holds {@code null}, never null
+     */
+    default <U> ObservableValue<U> map(Function<? super T, ? extends U> mapper) {
+        return new MappedBinding<>(this, mapper);
+    }
+
+    /**
+     * Returns an {@link ObservableValue} which provides a mapping of the value
+     * held by this {@code ObservableValue}, or {@code constant} when this
+     * {@code ObservableValue} holds {@code null}.
+     *
+     * @param constant an alternative value to use when this {@code ObservableValue}
+     *     holds {@code null}, can be null
+     * @return an {@link ObservableValue} which provides a mapping of the value
+     *     held by this {@code ObservableValue}, or {@code constant} when this
+     *     {@code ObservableValue} holds {@code null}, never null
+     */
+    default ObservableValue<T> orElse(T constant) {
+        return new OrElseBinding<>(this, constant);
+    }
+
+    /**
+     * Returns an {@link ObservableValue} which provides the value in the {@code
+     * ObservableValue} given by applying {@code mapper} on the value held by this
+     * {@code ObservableValue}, and is {@code null} when this
+     * {@code ObservableValue} holds {@code null}.<p>
+     *
+     * Returning {@code null} from {@code mapper} will result in an
+     * {@code ObservableValue} which holds {@code null}.
+     *
+     * @param <U> the type of values held by the resulting {@code ObservableValue}
+     * @param mapper a {@link Function} which converts a given value to an
+     *     {@code ObservableValue}, cannot be null
+     * @return an {@link ObservableValue} which provides the value in the
+     *     {@code ObservableValue} given by applying {@code mapper} on the value
+     *     held by this {@code ObservableValue}, and is {@code null} when this
+     *     {@code ObservableValue} holds {@code null}, never null
```

Comment:
```
an {@code ObservableValue} that holds the value of the {@code ObservableValue} resulting
from applying a mapping on the value held by this {@code ObservableValue}; never {@code null} itself
```

Reply:
How about:

    an {@code ObservableValue} holding the value of an {@code ObservableValue}
    resulting from a mapping of this {@code ObservableValue}'s value or
    holds {@code null} when the value is {@code null}; never returns {@code null}

They are tough to describe; if you don't like the `or holds {@code null}` parts I can remove those from all the functions.

Reply:
Here's another attempt:
```
a new {@link ObservableValue} instance that holds the value that was obtained by
applying the mapping function on the value held by this {@code ObservableValue}.
If this {@code ObservableValue} holds {@code null}, the new {@code ObservableValue}
will also hold {@code null}.
```